### ISSUE TITLE
Make `A.partition` types aware of type guards

### ISSUE THREAD

**Issue by bryanjswift** [mobily/ts-belt#25]
It would be a lovely addition if `partition` were aware of type guard functions. Something like this

```
export function partition<A>(
  xs: ReadonlyArray<A>,
  predicate: (value: A) => boolean
): [A[], A[]];

export function partition<A, B extends A>(
  xs: ReadonlyArray<A>,
  predicate: (value: A) => value is B
): [B[], Exclude<A, B>[]];
```

I thought it might be sufficient to drop this in https://github.com/mobily/ts-belt/blob/master/src/Array/Array.ts but I'm not confident enough to open it as a PR without being able to test locally and the tooling for this project is new to me so I couldn't afford to invest the time in getting it set up yet.

**Comment by mobily** [comment]
> It would be a lovely addition if partition were aware of type guard functions. 

fully agree, I will look into it! 🚀

**Comment by mobily** [comment]
@bryanjswift fixed in `v3.9.1` 🚀

- types: https://github.com/mobily/ts-belt/blob/6d8a02c74c19ef2e0cc3afc20d66d02dad55f34a/src/Array/index.ts#L443
- tests: https://github.com/mobily/ts-belt/blob/master/__tests__/Array/partition.test.ts
- changelog: https://mobily.github.io/ts-belt/docs/changelog
